tests/arch/arm/arm_no_multithreading/src/main.c

```diff
@@ -78,9 +78,16 @@ void test_main(void)
 
 			if (NVIC_GetPendingIRQ(i)) {
 				/* If the NVIC line is pending, it is
-				 * guaranteed that it is implemented.
+				 * guaranteed that it is implemented; clear the
+				 * line.
 				 */
-				break;
+				NVIC_ClearPendingIRQ(i);
+
+				if (!NVIC_GetPendingIRQ(i)) {
+					/* The NVIC line is cleared; use it. */
+					NVIC_SetPendingIRQ(i);
```

Comment:
here seems you are pending the line again, so why did you need to un-pend it earlier?

Reply:
See above.